intermittent_tracker/static/index.html

```diff
@@ -0,0 +1,333 @@
+                    }
+                    for (const key of keys) {
+                        if (uniques[key].size == 1) {
+                            allSame[key] = [...uniques[key]][0];
+                            allSameKeys.push(key);
+                        }
+                    }
+                    this.allSame = allSame;
+                    this.allSameKeys = allSameKeys;
+                } catch (e) {
+                    console.error(e);
+                } finally {
+                    setTimeout(this.update, 1000);
+                }
+            },
+            attemptsLimited() {
+                if (this.moreLinkVisible())
+                    return this.attempts.slice(0, this.liveRowLimit);
+                return this.attempts;
+            },
+            moreLinkVisible() {
+                return this.attempts != null
+                    && this.attempts.length > this.liveRowLimit
+                    && !this.more;
+            },
+            columnVisible(key) {
+                // hide any columns that we are filtering by
+                if (FILTER_KEYS.includes(key) && this.filterKeys.includes(key))
+                    return false;
+
+                // hide subtest column, unless path column is hidden
+                if (key == "subtest" && this.columnVisible("path"))
+                    return false;
+
+                // hide bulky columns, unless path and subtest are hidden
+                if (BULKY_KEYS.includes(key) && ["path", "subtest"].some(k => this.columnVisible(k)))
+                    return false;
+
+                // hide any columns that are also in “all results have the same”
+                return !(key in this.allSame);
+            },
+            filter(key, attempt) {
+                const params = new URLSearchParams(location.search);
+                params.set(key, attempt[key]);
+                return `?${params}`;
+            },
+            unfilter(key) {
+                const params = new URLSearchParams(location.search);
+                params.delete(key);
+                return `?${params}`;
+            },
+            short(url) {
+                return url != null ? url.split("/").at(-1) : null;
+            },
+        },
+    };
+
+    const RelativeTime = {
+        template: "#RelativeTime",
+        props: {
+            t: {type: Number, required: true},
+            relativeTo: {type: Number, required: true},
+        },
+        computed: {
+            iso() {
+                return new Date(this.t * 1000).toISOString();
+            },
+            text() {
+                const now = this.relativeTo;
+                return now - this.t >= 86400 ? `${Math.trunc((now - this.t) / 86400)}d`
+                    : now - this.t >= 3600 ? `${Math.trunc((now - this.t) / 3600)}h`
+                    : now - this.t >= 60 ? `${Math.trunc((now - this.t) / 60)}m`
+                    : `${Math.trunc(now - this.t)}s`;
+            },
+        },
+    };
```

Comment:
Would it be worth adding an " ago" here just to make the time display a little clearer? This sort of threw me off a little the first time I saw it. For instance, in Spain 15h means 3:00pm. :smile:

Reply:
huh, i had no idea! done in 200011061d599cc3edec7e237a1cb1ed9f2f211a